### DISCUSSION TITLE
Question/Help Needed: How to move all locally stored files to s3

### DISCUSSION THREAD

**Discussion by mwakaambrose** [pocketbase/pocketbase#2700] [Q&A] [answered]
Hello here, I recently ran out of server space and got s3 storage. I there a way or migration command that I can run in pocketbase to move all locally stored files to s3? I ask this because the server is full and causing issues now. It also won't upload new files to s3.

Please help!

**Comment by ganigeorgiev** [comment, marked as answer]
There is no builtin command for transferring your local files to an s3 storage but there are existing 3rd party tools that you can use like [rclone](https://rclone.org/), [s5md](https://github.com/peak/s5cmd), etc.

> It also won't upload new files to s3.

Is there any error that is returned? Are you sure that the S3 configuration is correct?

**Reply by mwakaambrose** [reply]
Hi @ganigeorgiev Thank you for the recommend tools, I'll check them out and report back. For this
> Is there any error that is returned? Are you sure that the S3 configuration is correct?

The error is generic as seen below
```
I/flutter (20846): Kijani: Failed to upload image to PocketBase: ClientException: {url: http://kijani.theonehq.com/api/collections/tasks/records, isAbort: false, statusCode: 500, response: {}, originalError: null}
```
and yes s3 is configured correctly because if you free up server space, it uploads to s3.

**Reply by ganigeorgiev** [reply]
We usually don't throw 500 error. Do you have a reverse proxy (eg. nginx) or something else that could interfere with the request?

**Reply by mwakaambrose** [reply]
Yes, pocketbase is sitting behind nginx. And yes, I am using nginx as a reverse proxy. Let me see if I can dig into nginx logs and come up with something.

**Reply by ganigeorgiev** [reply]
When uploading new files if the body request exceed 32MB it will store it in a temp location on the disk instead on the memory, so I guess it is possible if you are uploading large files to fail in case you don't have enough disk space.

**Reply by mwakaambrose** [reply]
I am actually uploading images, the largest image on the server is 6MB and the largest file on the server home is actually the pocketbase binary its self so I don't that is it. Let me get nginx error logs

**Reply by mwakaambrose** [reply]
Log file is empty for todays date.

![Screenshot 2023-06-12 at 22 23 20](https://github.com/pocketbase/pocketbase/assets/8328295/816d9fd7-e4ac-40ed-a2a9-19badcb946b3)

**Reply by ganigeorgiev** [reply]
What version of the Dart SDK do you use? Note that with v0.4.1+ the default client-side exception code was changed from `500` to `0`, so if 500 is not from nginx, then I suspect that you are using <= v0.4.0?

**Reply by mwakaambrose** [reply]
Hey @ganigeorgiev I used rclone and moved all files and everything seems to work great! Thank you for your help. Cheers!

**Comment by mwakaambrose** [comment]
@ganigeorgiev So when using `rclone`, I have to clone whats inside `./pb_data/storage/` and pocketbase will just map the paths like its on local disk or there are extra steps to do in pocketbase like a path change or something?

**Reply by ganigeorgiev** [reply]
No, there are no extra steps. You need only to copy the content of the `storage` directory as it is in the S3 configured bucket.

**Reply by mwakaambrose** [reply]
Got it! Thank you.

**Reply by mwakaambrose** [reply]
Hey @ganigeorgiev I used rclone and moved all files and everything seems to work great! Thank you for your help
